### PR TITLE
[unit-tests-only] Add unit tests for Avatar component

### PR DESCRIPTION
**Pull request by saw-jan**
## Description
Adds unit  tests for `Avatar` component

## Related Issue
- Part of #5234

## Motivation and Context
- increase unit tests coverage

## How Has This Been Tested?
- test environment: local & CI

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:
- [ ] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: packages/web-runtime/tests/unit/components/Avatar.spec.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
+import { shallowMount, createLocalVue } from '@vue/test-utils'
+import Vuex from 'vuex'
+import DesignSystem from 'owncloud-design-system'
+
+import Avatar from '../../../src/components/Avatar.vue'
```
Comment (kiranparajuli589) [inline packages/web-runtime/tests/unit/components/Avatar.spec.js:5]:
```suggestion
import Avatar from '@files/src/components/Avatar.vue'
```

File: packages/web-runtime/tests/unit/components/Avatar.spec.js [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+    ok: false,
+    status: 404
+  })
+)
+
+const ocSpinner = 'oc-spinner-stub'
+const ocAvatar = 'oc-avatar-stub'
+
+describe('Avatar component', () => {
+  describe('when loading is true', () => {
```
Comment (kiranparajuli589) [inline packages/web-runtime/tests/unit/components/Avatar.spec.js:28]:
```suggestion
  describe('when loading is set as true', () => {
```

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm 👍🏼

File: packages/web-runtime/tests/unit/components/Avatar.spec.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+      expect(spinner.exists()).toBeFalsy()
+      expect(avatar.exists()).toBeTruthy()
+    })
+    it('should set props on oc-avatar component', () => {
+      const avatar = wrapper.find(ocAvatar)
+
+      expect(avatar.props().width).toEqual(propsData.width)
+      expect(avatar.props().userName).toEqual(propsData.userName)
+      expect(avatar.props().src).toEqual('')
```
Comment (pascalwengerter) [inline packages/web-runtime/tests/unit/components/Avatar.spec.js:60]: Please also test the case where `src` is set and check that the `setUser` method is being called ;)

File: packages/web-runtime/tests/unit/components/Avatar.spec.js [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+function createStore() {
+  return new Vuex.Store({
+    getters: {
+      getToken: jest.fn(() => 'GFwHKXdsMgoFwt'),
+      capabilities: jest.fn(() => ({
+        files_sharing: {
+          user: {
+            profile_picture: true
```
Comment (pascalwengerter) [inline packages/web-runtime/tests/unit/components/Avatar.spec.js:85]: Please check the `Avatar component` and have a test for `profile_picture` being both set to true and to false (and checking it causes the desired behavior)!

**Review by pascalwengerter — CHANGES_REQUESTED** [review]
Looks good so far but I'd expect it to test some more functionaity 🤓

File: packages/web-runtime/tests/unit/components/Avatar.spec.js (hunk 4 above)
Comment (pascalwengerter) [reply packages/web-runtime/tests/unit/components/Avatar.spec.js:85]: Edit: Could also happen in an integration test I suppose. For the unit test it should be enough to spy & check that the `setUser` method is successfully being called on mounted

**Review by pascalwengerter — APPROVED** [review]
LGTM

**Pull request merged**